src/preprocessing.py

```diff
@@ -86,6 +86,43 @@ def preprocessing(df, country_metric_df):
     df = df.drop_duplicates()
     return df
 
+def create_security_test_features(api_df):
+    """ Create new features corresponding to security tests. The following features are being created:
+    - security_test_conducted (0: No, 1: Yes)
+    - 6 columns corresponding to each security test (0: PASS, 1: FAIL, 0.5: NOT AVAILABLE)
+    Parameters
+    ----------
+    api_df: Pandas Dataframe
+        Input dataframe with raw data columns, including security_test_results
+    Returns
+    -------
+    feat_security_tests_df : Pandas Dataframe
+        dataframe with endpoint_id and new security test features
+    """
+    # Data extraction and cleaning
+    security_test_df = api_df[["api_endpoint_id", "security_test_category", "security_test_result (FALSE=Passed; TRUE=Failed)"]]
+    security_test_df = security_test_df.rename(columns = {"api_endpoint_id":"endpoint_id",
+                                   "security_test_category":"test_category",
+                                    "security_test_result (FALSE=Passed; TRUE=Failed)":"test_result"})
+
+    # Create security_test_conducted feature
+    feat_security_df = security_test_df.groupby('endpoint_id').count()
+    feat_security_df["security_test_conducted"] = 0
+    feat_security_df.loc[feat_security_df["test_category"] > 0 , "security_test_conducted"] = 1
+    feat_security_df.drop(columns=["test_category", "test_result"], inplace=True)
+    
+    # Create test_result features
+    stacked_df = security_test_df.groupby(["endpoint_id", "test_category"]).mean()
```

Comment:
What is the purpose of this dataframe? This reduce the number of endpoint to 15, then how can we use it?